dirty_cat/_fuzzy_join.py

```diff
@@ -7,32 +7,171 @@
+        The columns of the main table.
+    aux : :class:`~pandas.DataFrame`
+        Another table with numerical columns.
+    aux_cols : str or list
+        The columns of the aux table.
+
+    Returns
+    -------
+    array-like
+        An array of the encoded columns of the main table.
+    array-like
+        An array of the encoded columns of the aux table.
+    """
+    aux_array = aux[aux_cols].to_numpy()
+    main_array = main[main_cols].to_numpy()
+    # Re-weighting to avoid measure specificity
+    scaler = StandardScaler()
+    scaler.fit(np.vstack((aux_array, main_array)))
+    aux_array = scaler.transform(aux_array)
+    main_array = scaler.transform(main_array)
+    return main_array, aux_array
+
+
+def _string_encoding(
+    main: pd.DataFrame,
+    main_cols: Union[list, str],
+    aux: pd.DataFrame,
+    aux_cols: Union[list, str],
+    analyzer: Literal["word", "char", "char_wb"],
+    ngram_range: Tuple[int, int],
+    encoder: Union[Literal["hashing"], _VectorizerMixin] = "hashing",
+) -> tuple:
+    """Encoding string columns.
+
+    Parameters
+    ----------
+    main : :class:`~pandas.DataFrame`
+        A table with string columns.
+    main_cols : str or list
+        The columns of the main table.
+    aux : :class:`~pandas.DataFrame`
+        Another table with string columns.
+    aux_cols : str or list
+        The columns of the aux table.
+    analyzer : {"word", "char", "char_wb"}
+        Analyzer parameter for the HashingVectorizer passed to
+        the encoder and used for the string similarities.
+        See fuzzy_join's docstring for more information.
+    ngram_range : int 2-tuple, optional, default=(2, 4)
+        The lower and upper boundary of the range of n-values for different
+        n-grams used in the string similarity.
+        See fuzzy_join's docstring for more information.
+    encoder: "hashing" or vectorizer instance, default="hashing"
+        Encoder parameter for the Vectorizer.
+        See fuzzy_join's docstring for more information.
+
+    Returns
+    -------
+    array-like
+        An array of the encoded columns of the main table.
+    array-like
+        An array of the encoded columns of the aux table.
+    """
+    # Make sure that the column types are string and categorical:
+    main_cols_clean = main[main_cols].astype(str)
+    aux_cols_clean = aux[aux_cols].astype(str)
+
+    if isinstance(main_cols, list) and isinstance(aux_cols, list):
+        first_col, other_cols = main_cols[0], main_cols[1:]
+        main_cols_clean = main_cols_clean[first_col].str.cat(
+            main_cols_clean[other_cols], sep="  "
+        )
+        first_col_aux, other_cols_aux = aux_cols[0], aux_cols[1:]
+        aux_cols_clean = aux_cols_clean[first_col_aux].str.cat(
+            aux_cols_clean[other_cols_aux], sep="  "
+        )
+    all_cats = pd.concat([main_cols_clean, aux_cols_clean], axis=0).unique()
+
+    if isinstance(encoder, str) and encoder == "hashing":
```

Comment:
Overall I'm -1 for introducing `encoder = "hashing"` by default instead of `None` because it introduces unnecessary complexity and a mixed argument signature.

Reply:
You can open an issue on this and we can discuss it more. I think this was done so that people don't think that we use the CountVectorizer..

Reply:
In that scenario, we could document in the docstring that the default `None` choice will fall back on `HashingVectorizer`. This is what is done in scikit-learn's ensemble estimator `StackingRegressor` for example:
https://github.com/scikit-learn/scikit-learn/blob/559609fe98ec2145788133687e64a6e87766bc77/sklearn/ensemble/_stacking.py#L788-L790

Reply:
Oh I see the point, actually this makes much more sense. The docstring is there to inform the user what's in the function, not the parameter name. I think I will change this now then.